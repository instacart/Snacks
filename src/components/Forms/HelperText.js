import React from 'react'
import PropTypes from 'prop-types'
import { colors } from '../../styles'

const style = {
  fontSize: '12px',
  lineHeight: '1.33',
  textAlign: 'right',
  color: colors.GRAY_46,
  width: '100%',
}

const HelperText = ({ helperText }) => (helperText ? <div css={style}>{helperText}</div> : null)

HelperText.propTypes = {
  /** Text */
  helperText: PropTypes.string,
}

export default HelperText
