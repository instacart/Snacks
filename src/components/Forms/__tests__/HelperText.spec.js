import React         from 'react'
import renderer      from 'react-test-renderer'
import { StyleRoot } from 'radium'
import HelperText    from '../HelperText'

it('renders HelperText correctly when floated', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <HelperText helperText="helperText Text" />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
