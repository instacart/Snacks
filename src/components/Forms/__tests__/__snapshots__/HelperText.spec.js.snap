// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HelperText correctly when floated 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      style={
        Object {
          "color": "#757575",
          "fontSize": "12px",
          "lineHeight": "1.33",
          "textAlign": "right",
          "width": "100%",
        }
      }
    >
      helperText Text
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
